### ISSUE TITLE
How to translate yolo4.cfg to a keras model properly

### ISSUE THREAD

**Issue by emadboctorx** [AlexeyAB/darknet#5743]
If something doesn’t work for you, then show 2 screenshots:
1. screenshots of your issue
2. screenshots with such information
```
./darknet detector test cfg/coco.data cfg/yolov4.cfg yolov4.weights data/dog.jpg
 CUDA-version: 10000 (10000), cuDNN: 7.4.2, CUDNN_HALF=1, GPU count: 1
 CUDNN_HALF=1
 OpenCV version: 4.2.0
 0 : compute_capability = 750, cudnn_half = 1, GPU: GeForce RTX 2070
net.optimized_memory = 0
mini_batch = 1, batch = 8, time_steps = 1, train = 0
   layer   filters  size/strd(dil)      input                output
   0 conv     32       3 x 3/ 1    608 x 608 x   3 ->  608 x 608 x  32 0.639 BF
```

If you do not get an answer for a long time, try to find the answer among Issues with a Solved label: https://github.com/AlexeyAB/darknet/issues?q=is%3Aopen+is%3Aissue+label%3ASolved

A few days ago, I added an implementation to yolov3 in tensorflow which you can see [here](https://github.com/emadboctorx/yolov3-keras-tf2) I'm currently working on adding a v4 configuration as well however when I'm  reading the yolo4.cfg, I can't seem to understand how the route layers work and what ex: [route] layers=85 or [route] layers=-1 -16 ...

I convert the cfg file to a format that suits my code that looks like the following(yolo3 configuration)

```
conv,32,3,1,True

conv,64,3,2,True

conv,32,1,1,True,append

conv,64,3,1,True,add

conv,128,3,2,True

conv,64,1,1,True,append

conv,128,3,1,True,add

conv,64,1,1,True,append

conv,128,3,1,True,add

conv,256,3,2,True

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

conv,128,1,1,True,append

conv,256,3,1,True,add

skip_36

conv,512,3,2,True

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

conv,256,1,1,True,append

conv,512,3,1,True,add

skip_61

conv,1024,3,2,True

conv,512,1,1,True,append

conv,1024,3,1,True,add

conv,512,1,1,True,append

conv,1024,3,1,True,add

conv,512,1,1,True,append

conv,1024,3,1,True,add

conv,512,1,1,True,append

conv,1024,3,1,True,add

conv,512,1,1,True

conv,1024,3,1,True

conv,512,1,1,True

conv,1024,3,1,True

conv,512,1,1,True

detection_0

output_0,512

conv,256,1,1,True

upsample

concat,skip_61

conv,256,1,1,True

conv,512,3,1,True

conv,256,1,1,True

conv,512,3,1,True

conv,256,1,1,True

detection_1

output_1,256

conv,128,1,1,True

upsample

concat,skip_36

conv,128,1,1,True

conv,256,3,1,True

conv,128,1,1,True

conv,256,3,1,True

conv,128,1,1,True

detection_2

output_2,128

training_model

boxes_0

boxes_1

boxes_2

nms

inference_model
```

add and append are for indicating shortcut layers. Can you please help me understand how the route layer indications work so I can make the proper model configuration?

Here's what I did so far which I think is incorrect

```
conv,32,3,1,True,,mish

conv,64,3,2,True,,mish

conv,64,1,1,True,,mish

skip_3

conv,64,1,1,True,,mish

conv,32,1,1,True,append,mish

conv,64,3,1,True,add,mish

conv,64,1,1,True,,mish

concat,skip_3

conv,64,1,1,True,,mish

conv,128,3,2,True,,mish

conv,64,1,1,True,,mish

skip_10

conv,64,1,1,True,,mish

conv,64,1,1,True,append,mish

conv,64,3,1,True,add,mish

conv,64,1,1,True,append,mish

conv,64,3,1,True,add,mish

conv,64,1,1,True,,mish

concat,skip_10

conv,128,1,1,True,,mish

conv,256,3,2,True,,mish

conv,128,1,1,True,,mish

skip_19

conv,128,1,1,True,,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,append,mish

conv,128,3,1,True,add,mish

conv,128,1,1,True,,mish

concat,skip_19

conv,256,1,1,True,,mish

conv,512,3,2,True,,mish

conv,256,1,1,True,,mish

skip_40

conv,256,1,1,True,,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,append,mish

conv,256,3,1,True,add,mish

conv,256,1,1,True,,mish

concat,skip_40

conv,512,1,1,True,,mish

conv,1024,3,2,True,,mish

conv,512,1,1,True,,mish

skip_61

conv,512,1,1,True,,mish

conv,512,1,1,True,append,mish

conv,512,3,1,True,add,mish

conv,512,1,1,True,append,mish

conv,512,3,1,True,add,mish

conv,512,1,1,True,append,mish

conv,512,3,1,True,add,mish

conv,512,1,1,True,append,mish

conv,512,3,1,True,add,mish

conv,512,1,1,True,,mish

concat,skip_61

conv,1024,1,1,True,,mish

conv,512,1,1,True,,leaky

conv,1024,3,1,True,,leaky

conv,512,1,1,True,,leaky

skip_75

maxpool,13,1

skip_76

maxpool,9,1

skip_77

maxpool,5,1

concat,skip_77,skip_76,skip_75

conv,512,1,1,True,,leaky

conv,1024,3,1,True,,leaky

conv,512,1,1,True,,leaky

skip_80

conv,256,1,1,True,,leaky

upsample

skip_82

conv,256,1,1,True,,leaky

concat,skip_82

conv,256,1,1,True,,leaky

conv,512,3,1,True,,leaky

conv,256,1,1,True,,leaky

conv,512,3,1,True,,leaky

conv,256,1,1,True,,leaky

skip_88

conv,128,1,1,True,,leaky

upsample

skip_89

conv,128,1,1,True,,leaky

concat,skip_89

conv,128,1,1,True,,leaky

conv,256,3,1,True,,leaky

conv,128,1,1,True,,leaky

conv,256,3,1,True,,leaky

conv,128,1,1,True,,leaky

detection_0

output_0,128

skip_95

conv,256,3,2,True,,leaky

concat,skip_88

conv,256,1,1,True,,leaky

conv,512,3,1,True,,leaky

conv,256,1,1,True,,leaky

conv,512,3,1,True,,leaky

conv,256,1,1,True,,leaky

detection_1

output_1,256

skip_101

conv,512,3,2,True,,leaky

concat,skip_80

conv,512,1,1,True,,leaky

conv,1024,3,1,True,,leaky

conv,512,1,1,True,,leaky

conv,1024,3,1,True,,leaky

conv,512,1,1,True,,leaky

detection_2

output_2,512

training_model
```
